File: app/src/main/java/com/example/vishistvarugeese/memory/Repositories/NoteRepository.java
package com.example.vishistvarugeese.memory.Repositories;

import android.annotation.SuppressLint;
import android.arch.lifecycle.LiveData;
import android.content.Context;
import android.os.AsyncTask;
import android.util.Log;

import com.example.vishistvarugeese.memory.Helper.MemoryPlusDatabase;
import com.example.vishistvarugeese.memory.Variables.Entity.Note;

import java.util.List;

public class NoteRepository {

    @SuppressLint("StaticFieldLeak")
    public void insertNote(final Note note, final Context context) {

        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... voids) {
                MemoryPlusDatabase.getInstance(context).getNoteDAO().insert(note);
                return null;
            }
        }.execute();
    }

    @SuppressLint("StaticFieldLeak")
    public void updateNote(final String title, final String description, final String timestamp, final int position, final Context context) {

        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... voids) {
                MemoryPlusDatabase.getInstance(context).getNoteDAO().update(title, description, timestamp, position);
                return null;
            }
        }.execute();
    }

    @SuppressLint("StaticFieldLeak")
    public void deleteAll(final Context context) {
        new AsyncTask<Void, Void, Void>() {
            @Override
            protected Void doInBackground(Void... voids) {
                MemoryPlusDatabase.getInstance(context).getNoteDAO().deleteAll();
                return null;
            }
        }.execute();
    }

    @SuppressLint("StaticFieldLeak")
    public void deleteNote(final int id, final Context context) {

            new AsyncTask<Void, Void, Void>() {
                @Override
                protected Void doInBackground(Void... voids) {
                    MemoryPlusDatabase.getInstance(context).getNoteDAO().delete(id);
                    return null;
                }
            }.execute();
    }

    public LiveData<List<Note>> getAllNotes(Context context) {
        return MemoryPlusDatabase.getInstance(context).getNoteDAO().getAllNotes();
    }




}

File: app/src/main/java/com/example/vishistvarugeese/memory/Repositories/NoteRepository.kt
package com.example.vishistvarugeese.memory.Repositories

import android.annotation.SuppressLint
import android.arch.lifecycle.LiveData
import android.content.Context
import android.os.AsyncTask
import com.example.vishistvarugeese.memory.Helper.MemoryPlusDatabase
import com.example.vishistvarugeese.memory.Variables.Entity.Note

class NoteRepository {
    @SuppressLint("StaticFieldLeak")
    fun insertNote(note: Note?, context: Context) {
        object : AsyncTask<Void?, Void?, Void?>() {
            protected override fun doInBackground(vararg voids: Void): Void? {
                MemoryPlusDatabase.Companion.getInstance(context).getNoteDAO().insert(note)
                return null
            }
        }.execute()
    }

    @SuppressLint("StaticFieldLeak")
    fun updateNote(title: String?, description: String?, timestamp: String?, position: Int, context: Context) {
        object : AsyncTask<Void?, Void?, Void?>() {
            protected override fun doInBackground(vararg voids: Void): Void? {
                MemoryPlusDatabase.Companion.getInstance(context).getNoteDAO().update(title, description, timestamp, position)
                return null
            }
        }.execute()
    }

    @SuppressLint("StaticFieldLeak")
    fun deleteAll(context: Context) {
        object : AsyncTask<Void?, Void?, Void?>() {
            protected override fun doInBackground(vararg voids: Void): Void? {
                MemoryPlusDatabase.Companion.getInstance(context).getNoteDAO().deleteAll()
                return null
            }
        }.execute()
    }

    @SuppressLint("StaticFieldLeak")
    fun deleteNote(id: Int, context: Context) {
        object : AsyncTask<Void?, Void?, Void?>() {
            protected override fun doInBackground(vararg voids: Void): Void? {
                MemoryPlusDatabase.Companion.getInstance(context).getNoteDAO().delete(id)
                return null
            }
        }.execute()
    }

    fun getAllNotes(context: Context): LiveData<List<Note>> {
        return MemoryPlusDatabase.Companion.getInstance(context).getNoteDAO().getAllNotes()
    }
}